.github/workflows/build-docker-image-and-binaries.yaml

```diff
@@ -63,6 +63,39 @@ jobs:
               TOFND_PATH="/Users/runner/work/tofnd/tofnd/target/release"
           fi
 
+      - name: build darwin binaries
+        env:
+          SEMVER: ${{ github.event.inputs.tag }}
+        if: matrix.os == 'macos-latest'
+        run: |
+          OS="darwin"
+          ARCH="${{ matrix.arch }}"
+          if [ "$ARCH" == "arm64" ]
+          then
+            export LIBRARY_PATH=/opt/homebrew/lib:$LIBRARY_PATH
+            export LD_LIBRARY_PATH=/opt/homebrew/lib:$LD_LIBRARY_PATH
+            export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
+            brew uninstall --ignore-dependencies gmp
+            ARM_DEPENDENCY=$(brew fetch --force --bottle-tag=arm64_sonoma gmp | grep Downloaded | awk '{print $3}')
+            brew install "$ARM_DEPENDENCY"
+            rustup target add aarch64-apple-darwin
+            cargo build --release --target aarch64-apple-darwin
+            mkdir -p tofndbin
+            mv /Users/runner/work/tofnd/tofnd/target/aarch64-apple-darwin/release/tofnd "./tofndbin/tofnd-$OS-$ARCH-$SEMVER"
+          else
+            export RUSTFLAGS="-L/opt/homebrew/lib"
+            export LIBRARY_PATH=/opt/homebrew/lib:$LIBRARY_PATH
+            export LD_LIBRARY_PATH=/opt/homebrew/lib:$LD_LIBRARY_PATH
+            export C_INCLUDE_PATH=/opt/homebrew/include:$C_INCLUDE_PATH
+            export CPLUS_INCLUDE_PATH=/opt/homebrew/include:$CPLUS_INCLUDE_PATH
+            export HOMEBREW_NO_INSTALL_CLEANUP=TRUE
```

Comment:
Duplication between the two if statements. Run the common cmds at the top

Reply:
done